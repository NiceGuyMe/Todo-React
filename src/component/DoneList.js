export function DoneList({ doneList }) {
  return (
    <ul>
      {doneList.map((e, index) => (
        <li key={index}>{e}</li>
      ))}
    </ul>
  );
}
