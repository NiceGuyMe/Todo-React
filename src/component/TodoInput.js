import { useState } from "react";

export function TodoInput({ onAddTodo }) {
  const [todo, setTodo] = useState("");

  const handleChange = (event) => {
    setTodo(event.target.value);
  };

  const handleKeyDown = (e) => {
    if (e.key === "Enter") {
      onAddTodo(todo);
      setTodo("");
    }
  };

  return (
    <input
      placeholder="Enter todo"
      className="input"
      type="text"
      value={todo}
      onChange={handleChange}
      onKeyDown={handleKeyDown}
    />
  );
}
