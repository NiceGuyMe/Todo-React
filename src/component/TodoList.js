export function TodoList({ todoList, onCheckboxChange }) {
  return (
    <ul>
      {todoList.map((e, index) => (
        <li key={index}>
          {e}
          <input
            className="checkbox"
            type="checkbox"
            checked={false}
            onChange={() => onCheckboxChange(index)}
          />
        </li>
      ))}
    </ul>
  );
}
