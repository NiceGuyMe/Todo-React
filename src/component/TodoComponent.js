import { useState } from "react";

import { TodoInput } from "./TodoInput";
import { TodoList } from "./TodoList";
import { DoneList } from "./DoneList";

export function TodoApp() {
  const [todoList, setTodoList] = useState([]);

  const [doneList, setdoneList] = useState([]);

  const addToTodoList = (todo) => {
    if (todo !== "") {
      setTodoList([...todoList, todo]);
    } else {
      throw new Error("todo is empty");
    }
  };

  const handleCheckboxChange = (index) => {
    const checkedItem = todoList[index];
    setdoneList([...doneList, checkedItem]);
    todoList.splice(index, 1);
    setTodoList([...todoList]);
  };

  return (
    <>
      <div className="todoinput">
        <TodoInput onAddTodo={addToTodoList} />
      </div>
      <div className="todoContainer">
        <div className="TodoList">
          <TodoList
            todoList={todoList}
            onCheckboxChange={handleCheckboxChange}
          />
        </div>
      </div>

      <div className="DoneContainer">
        <DoneList doneList={doneList} />
      </div>
    </>
  );
}
