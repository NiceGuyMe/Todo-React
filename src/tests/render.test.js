import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { TodoList } from "../component/TodoList";
import { DoneList } from "../component/DoneList";
import { TodoInput } from "../component/TodoInput";
import { render, fireEvent, screen } from "@testing-library/react";

test("Show the Done header", function () {
  render(<DoneList doneList={[]} />);
  expect(screen.getByText("DONE")).toBeInTheDocument();
});

test("Show the TODO header", () => {
  render(<TodoList todoList={[]} onCheckboxChange={() => {}} />);
  expect(screen.getByText("TODO")).toBeInTheDocument();
});

test("Show Todo lists", () => {
  const todoList = ["Item 1", "Item 2", "Item 3"];
  render(<TodoList todoList={todoList} onCheckboxChange={() => {}} />);
  todoList.forEach((todo) => {
    expect(screen.getByText(todo)).toBeInTheDocument();
  });
});

test("Show Done lists", function () {
  const todoList = ["Item 1", "Item 2", "Item 3"];
  render(<DoneList doneList={todoList} />);
  todoList.forEach((done) => {
    expect(screen.getByText(done)).toBeInTheDocument();
  });
});

test("Show input placeholder", function () {
  render(<TodoInput />);
  expect(screen.getByPlaceholderText("Enter todo")).toBeInTheDocument();
});

test("Clear input when press Enter", () => {
  const onAddTodo = jest.fn();
  render(<TodoInput onAddTodo={onAddTodo} />);
  const input = screen.getByPlaceholderText("Enter todo");
  fireEvent.change(input, { target: { value: "Test Todo" } });
  fireEvent.keyDown(input, { key: "Enter" });
  expect(input.value).toBe("");
});

test("calls onCheckboxChange when checkbox is clicked", () => {
  const onCheckboxChange = jest.fn();
  const todoList = ["Item 1", "Item 2", "Item 3"];
  render(<TodoList todoList={todoList} onCheckboxChange={onCheckboxChange} />);
  fireEvent.click(screen.getAllByRole("checkbox")[1]);
  expect(onCheckboxChange).toHaveBeenCalledWith(1);
});

test("calls onAddTodo when the enter key is pressed", () => {
  const onAddTodo = jest.fn();
  render(<TodoInput onAddTodo={onAddTodo} />);
  const input = screen.getByPlaceholderText("Enter todo");
  fireEvent.change(input, { target: { value: "Test Todo" } });
  fireEvent.keyDown(input, { key: "Enter" });
  expect(onAddTodo).toHaveBeenCalledWith("Test Todo");
});
