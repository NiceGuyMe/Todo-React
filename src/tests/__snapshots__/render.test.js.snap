// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot DoneList 1`] = `undefined`;

exports[`snapshot TodoApp 1`] = `undefined`;

exports[`snapshot TodoInput 1`] = `undefined`;

exports[`snapshot TodoList 1`] = `undefined`;
